gs-rest-client/src/main/java/client/SimpleClient.java

```diff
@@ -118,95 +118,132 @@ public static void fixMyBug(ServerRequest serverRequest, TokenizerBuilder bobThe
+                    " method> (fix, echo, index)");
             System.exit(0);
         }
 
         String fileName = args[0];
-        int startLine = Integer.parseInt(args[1]);
-        int endLine = Integer.parseInt(args[2]);
-        String method = args[3];
-
+        String errorMessage = args[1];
+        int startLine = Integer.parseInt(args[2]);
+        int endLine = Integer.parseInt(args[3]);
+        String method = args[4];
         ServerRequest serverRequest;
 
-        if (method.equals("index")) {
-            // Hack into the parameters of serverRequest the arguments for index
-            serverRequest = new ServerRequest(fileName, args[1]);
-            System.out.println(serverRequest.getBuggyCode());
-            fixMyBug(serverRequest, method);
-
-        } else {
-            String buggyCodeBlock = "";
-            try {
-                buggyCodeBlock = getLinesFromFile(fileName, startLine, endLine);
-            } catch (IOException e) {
-                e.printStackTrace();
-            }
-
-            TokenizerBuilder tokenizedCodeBlock = null;
-            //Tokenize the input file
-            try {
-                tokenizedCodeBlock = tokenize(buggyCodeBlock);
-            } catch (Exception ex) {
-                ex.printStackTrace();
-            }
+        //Identify the buggy code block within the provided file.
+        String buggyCodeBlock = "";
+        try {
+            buggyCodeBlock = getLinesFromFile(fileName, startLine, endLine);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        System.out.println("Buggy code block:\n\n" + buggyCodeBlock + "\n");
 
-            String errorMessage = "Custom-Error-Message";
-            StringBuilder stringBuilder = new StringBuilder();
-            //try {
-            stringBuilder.append(tokenizedCodeBlock.getString());
-            System.out.println("Buggy code:");
-            System.out.println(buggyCodeBlock);
-            System.out.println("\n");
-            //} catch (IOException e) {
-            //    e.printStackTrace();
-            //}
-
-            serverRequest = new ServerRequest(stringBuilder.toString(), errorMessage);
-            System.out.println("Sending tokenized code: " + serverRequest.getBuggyCode() + "\nWith the error message: " + serverRequest.getErrorMessage());
-            fixMyBug(serverRequest, tokenizedCodeBlock, method);
+        //Tokenize the buggy code block and return as a TokenizerBuilder
+        TokenizerBuilder tokenBuilder = null;
+        try {
+            tokenBuilder = tokenize(buggyCodeBlock);
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+        System.out.println("Tokens:\n\n" + tokenBuilder.getString() + "\n");
+
+        //Generate the serverRequest from the provided tokenized code
+        //and the input error message.
+        StringBuilder tokenizedCodeBlock = new StringBuilder();
```

Comment:
Here I name the StringBuilder "tokenizedCodeBlock," as it should contain a string of our tokenized code.